clients/csharp-dotnet-core/voice-assistant-test/tool/MainService.cs

```diff
@@ -206,80 +204,174 @@ private static async Task<bool> ProcessTestFiles(AppSettings appSettings)
                 // Keep track of high-level (summary) results for all dialogs in a single input test file. This list will be serialized to JSON as part of the overall single test report.
                 List<DialogReport> dialogReports = new List<DialogReport>();
 
-                foreach (Dialog dialog in fileContents)
+                bool sendFirst = true;
+                string outputType = string.Empty;
+                if (tests.WavAndUtterancePairs)
                 {
-                    Trace.IndentLevel = 1;
-                    if (dialog.Skip)
-                    {
-                        Trace.TraceInformation($"Skipping DialogID {dialog.DialogID}");
-                        continue;
-                    }
-                    else
+                    outputType = "Output-Wav";
+                    await ProcessDialogAndGenerateReport(outputType, outputPath, testPass, botConnector, dialogReports, connectionEstablished, dialogResults, fileContents, isFirstDialog, tests, testName, sendFirst, inputFileName, allInputFilesTestReport, outputFileName, appSettings).ConfigureAwait(false);
+
+                    sendFirst = false;
+                    outputType = "Output-Text";
+
+                    await ProcessDialogAndGenerateReport(outputType, outputPath, testPass, botConnector, dialogReports, connectionEstablished, dialogResults, fileContents, isFirstDialog, tests, testName, sendFirst, inputFileName, allInputFilesTestReport, outputFileName, appSettings).ConfigureAwait(false);
+
+#if USE_ARIA_LOGGING
+                    AriaLogger.Stop();
+#endif

```

Comment:
Move this down... Stop ARIA logging just before your return (see the original version before your changes):

#if USE_ARIA_LOGGING
            AriaLogger.Stop();
#endif
            return testPass;

Reply:
Done